apps/studio/src/pages/sites/[siteId]/settings/agency.tsx

```diff
@@ -37,6 +36,12 @@ const AgencySettingsPage: NextPageWithLayout = () => {
   })
   const trpcUtils = trpc.useUtils()
   const toast = useToast()
+  const agencySettingsSchema = Type.Pick(
+    SiteConfigSchema,
+    Type.Union([Type.Literal("siteName"), Type.Literal("agencyName")]),
+  )
```

Comment:
praise: I love this cleaner design tbh, still allows JSONForms to be the one doing the calculation and rendering

thought: I still think there is some value in having the components export some sort of function, say `getScopedSchema(scope)` that allows Studio to get the appropriate subschema to render. That function's implementation would be this `Type.Pick`, but at least the ownership is on the components rather than on Studio